packages/xo-web/src/xo-app/sr/tab-general.js

```diff
@@ -32,9 +168,97 @@ export default class TabGeneral extends Component {
     }
   }
 
-  render() {
-    const { sr, vdis, vdiSnapshots, unmanagedVdis } = this.props
+  _getDiskGroups = createSelector(
+    () => this.props.vdis,
+    () => this.props.vdiSnapshots,
+    () => this.props.unmanagedVdis,
+    (vdis, vdiSnapshots, unmanagedVdis) => {
+      const diskParents = []
+      const groups = []
+      const snapshotsByVdi = groupBy(vdiSnapshots, '$snapshot_of')
+      const unmanagedVdisById = keyBy(unmanagedVdis, 'id')
+      let baseCopiesUsage
+      let bc
+      let nBaseCopies
+      let root
+      // group VDIs by their initial base copy.
+      const disksByRoot = groupBy(
```

Comment:
For the sake of readability, I think you should `groupBy` in the next step or between the 2 steps.